File: bucket-water-oms-java/src/main/java/com/bucketwater/oms/module/customer/service/CustomerService.java
package com.bucketwater.oms.module.customer.service;

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;
import com.bucketwater.oms.module.customer.dto.CustomerDTO;
import com.bucketwater.oms.module.customer.entity.Customer;
import com.bucketwater.oms.module.customer.mapper.CustomerMapper;
import com.bucketwater.oms.module.station.entity.StationAccount;
import com.bucketwater.oms.module.station.mapper.StationAccountMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class CustomerService {

    @Autowired
    private CustomerMapper customerMapper;

    @Autowired
    private StationAccountMapper stationAccountMapper;

    public List<CustomerDTO> getCustomers(Long stationId, String keyword) {
        QueryWrapper<Customer> wrapper = new QueryWrapper<>();
        if (stationId != null) {
            wrapper.eq("station_id", stationId);
        }
        if (keyword != null && !keyword.isBlank()) {
            wrapper.and(w -> w.like("name", keyword).or().like("phone", keyword));
        }
        wrapper.eq("status", "active");
        wrapper.orderByDesc("created_at");

        List<Customer> customers = customerMapper.selectList(wrapper);
        return customers.stream().map(this::convertToDTO).collect(Collectors.toList());
    }

    public CustomerDTO getCustomerById(Long customerId) {
        Customer customer = customerMapper.selectById(customerId);
        if (customer == null) {
            return null;
        }
        return convertToDTO(customer);
    }

    public CustomerDTO createCustomer(Long stationId, String name, String phone, String address, String contact) {
        Customer customer = new Customer();
        customer.setStationId(stationId);
        customer.setName(name);
        customer.setPhone(phone);
        customer.setAddress(address);
        customer.setContact(contact);
        customer.setStatus("active");
        customerMapper.insert(customer);
        return convertToDTO(customer);
    }

    public CustomerDTO updateCustomer(Long customerId, String name, String phone, String address, String contact) {
        Customer customer = customerMapper.selectById(customerId);
        if (customer == null) {
            return null;
        }
        if (name != null) customer.setName(name);
        if (phone != null) customer.setPhone(phone);
        if (address != null) customer.setAddress(address);
        if (contact != null) customer.setContact(contact);
        customerMapper.updateById(customer);
        return convertToDTO(customer);
    }

    public void deleteCustomer(Long customerId) {
        Customer customer = customerMapper.selectById(customerId);
        if (customer != null) {
            customer.setStatus("deleted");
            customerMapper.updateById(customer);
        }
    }

    private CustomerDTO convertToDTO(Customer customer) {
        CustomerDTO dto = new CustomerDTO();
        dto.setId(customer.getId());
        dto.setName(customer.getName());
        dto.setPhone(customer.getPhone());
        dto.setAddress(customer.getAddress());
        dto.setContact(customer.getContact());
        dto.setStatus(customer.getStatus());
        dto.setCreatedAt(customer.getCreatedAt());

        if (customer.getStationId() != null) {
            StationAccount account = stationAccountMapper.selectOne(
                new QueryWrapper<StationAccount>().eq("station_id", customer.getStationId())
            );
            if (account != null) {
                dto.setBalance(account.getDepositBalance());
                dto.setCreditLimit(account.getCreditLimit());
                dto.setCreditUsed(account.getCreditUsed());
                dto.setOwedBuckets(account.getOwedBucketNum());
                dto.setThreshold(account.getOwedThreshold());
            }
        }

        return dto;
    }
}

File: bucket-water-oms-admin-mobile/lib/services/customer_service.dart
import '../core/network/api_client.dart';
import '../models/customer_model.dart';

class CustomerService {
  static final CustomerService _instance = CustomerService._internal();
  factory CustomerService() => _instance;
  CustomerService._internal();

  final ApiClient _apiClient = ApiClient();

  Future<List<CustomerModel>> getCustomers({
    String? stationId,
    String? keyword,
  }) async {
    final headers = <String, String>{};
    if (stationId != null) {
      headers['X-Station-Id'] = stationId;
    }

    final queryParams = <String, String>{};
    if (keyword != null && keyword.isNotEmpty) {
      queryParams['keyword'] = keyword;
    }

    final response = await _apiClient.get(
      '/customers',
      queryParams: queryParams,
      headers: headers,
    );

    if (response.success) {
      final data = response.data;
      if (data is List) {
        return data.map((e) => CustomerModel.fromJson(e)).toList();
      }
      return [];
    } else {
      throw ApiException(
        response.message ?? '获取客户列表失败',
        statusCode: response.code,
      );
    }
  }

  Future<CustomerModel> getCustomer(String customerId) async {
    final response = await _apiClient.get('/customers/$customerId');

    if (response.success) {
      return CustomerModel.fromJson(response.data);
    } else {
      throw ApiException(
        response.message ?? '获取客户详情失败',
        statusCode: response.code,
      );
    }
  }

  Future<CustomerModel> createCustomer({
    required String stationId,
    required String name,
    required String phone,
    String? address,
    String? contact,
  }) async {
    final response = await _apiClient.post(
      '/customers',
      body: {
        'name': name,
        'phone': phone,
        if (address != null) 'address': address,
        if (contact != null) 'contact': contact,
      },
      headers: {'X-Station-Id': stationId},
    );

    if (response.success) {
      return CustomerModel.fromJson(response.data);
    } else {
      throw ApiException(
        response.message ?? '创建客户失败',
        statusCode: response.code,
      );
    }
  }

  Future<CustomerModel> updateCustomer(
    String customerId, {
    String? name,
    String? phone,
    String? address,
    String? contact,
  }) async {
    final body = <String, dynamic>{};
    if (name != null) body['name'] = name;
    if (phone != null) body['phone'] = phone;
    if (address != null) body['address'] = address;
    if (contact != null) body['contact'] = contact;

    final response = await _apiClient.put(
      '/customers/$customerId',
      body: body,
    );

    if (response.success) {
      return CustomerModel.fromJson(response.data);
    } else {
      throw ApiException(
        response.message ?? '更新客户失败',
        statusCode: response.code,
      );
    }
  }

  Future<bool> deleteCustomer(String customerId) async {
    final response = await _apiClient.delete('/customers/$customerId');
    return response.success;
  }
}
